pandas/core/internals/blocks.py

```diff
@@ -1378,6 +1366,7 @@ def func(cond, values, other):
             if not (
                 (self.is_integer or self.is_bool)
                 and lib.is_scalar(other)
+                and lib.is_float(other)
```

Comment:
did this cause a change in tests? you can drop the first 2 conditions i think if you add this

Reply:
In another branch I found that the np.isnan check was raising a TypeError in some cases.  That doesn't really matter since ATM this is inside a try/except block, but this isnt the part of the code that we are expecting to raise, so being more explicit is better.

Will remove the unnecessary part of the check.

Reply:
updated;  de facto green